pkg/providers/nutanix/validator.go

```diff
@@ -156,6 +159,22 @@ func (v *Validator) validateFailureDomains(ctx context.Context, client Client, c
 				return err
 			}
 		}
+
+		for _, workerMachineGroupName := range fd.WorkerMachineGroups {
+			if err := v.validateWorkerMachineGroup(spec, workerMachineGroupName); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Validator) validateWorkerMachineGroup(spec *cluster.Spec, workerMachineGroupName string) error {
+	workerMachineGroupNames := getWorkerMachineGroupNames(spec)
+
+	if !sliceContains(workerMachineGroupNames, workerMachineGroupName) {
+		return fmt.Errorf("worker machine group %s not found in the cluster worker node group definitions", workerMachineGroupName)
```

Comment:
can you use a map here instead of a slice to check if the worker group name exists? That should give you faster lookups

Reply:
Fixed